lib/Common/Memory/Recycler.cpp

```diff
@@ -4464,12 +4500,16 @@ Recycler::FinishConcurrent()
 
         const BOOL forceFinish = flags & CollectOverride_ForceFinish;
 
+#if ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+        if (forceFinish || !(IsConcurrentExecutingState()))
```

Comment:
Isn't this the same condition as #else?

Reply:
good catch. fixed.

---
In reply to: [148170532](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148170532) [](ancestors = 148170532)